tools/Microsoft.AspNet.Server.Kestrel.GeneratedCode/FrameFeatureCollection.cs

```diff
@@ -25,26 +25,40 @@ public virtual void BeforeCompile(BeforeCompileContext context)
 
         public static string GeneratedFile()
         {
-            var commonFeatures = new[]
+            var alwaysFeatures = new[]
             {
                 typeof(IHttpRequestFeature),
                 typeof(IHttpResponseFeature),
                 typeof(IHttpRequestIdentifierFeature),
-                typeof(IHttpSendFileFeature),
                 typeof(IServiceProvidersFeature),
-                typeof(IHttpAuthenticationFeature),
                 typeof(IHttpRequestLifetimeFeature),
+                typeof(IHttpConnectionFeature)
+            };
+
+            var commonFeatures = new[]
+            {
+                typeof(IHttpAuthenticationFeature),
                 typeof(IQueryFeature),
-                typeof(IFormFeature),
+                typeof(IFormFeature)
+            };
+
+            var sometimesFeatures = new[]
+            {
+                typeof(IHttpUpgradeFeature),
```

Comment:
@lodejard The 6 features currently not included with objects and which will use the backing list are

``IResponseCookiesFeature``
``IItemsFeature``
``ITlsConnectionFeature``
``IHttpWebSocketFeature``
``ISessionFeature``
``IHttpSendFileFeature``

Do you believe this to be incorrect?